pkgs/record_use/lib/record_use_internal.dart

```diff
@@ -0,0 +1,8 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+export 'record_use.dart';
+export 'src/data_classes/definition.dart' show Definition;
```

Comment:
If `definition.dart` is internal it should be somehow obvious from the file path.

Please consider making
* `src/public/*.dart` & `src/internal/*.dart`
* `src/data_classes/public/*.dart` & `src/data_classes/internal/*.dart`

The `data_classes` directory isn't really needed.